protokube/pkg/gossip/dns/hosts/hosts.go

```diff
@@ -159,49 +158,49 @@ func pseudoAtomicWrite(p string, b []byte, mode os.FileMode) error {
 	}
 }
 
-func atomicWriteFile(filename string, data []byte, perm os.FileMode) error {
-	dir := filepath.Dir(filename)
-
-	tempFile, err := ioutil.TempFile(dir, ".tmp"+filepath.Base(filename))
-	if err != nil {
-		return fmt.Errorf("error creating temp file in %q: %v", dir, err)
-	}
-
-	mustClose := true
-	mustRemove := true
-
-	defer func() {
-		if mustClose {
-			if err := tempFile.Close(); err != nil {
-				klog.Warningf("error closing temp file: %v", err)
-			}
-		}
-
-		if mustRemove {
-			if err := os.Remove(tempFile.Name()); err != nil {
-				klog.Warningf("error removing temp file %q: %v", tempFile.Name(), err)
-			}
-		}
-	}()
-
-	if _, err := tempFile.Write(data); err != nil {
-		return fmt.Errorf("error writing temp file: %v", err)
-	}
-
-	if err := tempFile.Close(); err != nil {
-		return fmt.Errorf("error closing temp file: %v", err)
-	}
-
-	mustClose = false
-
-	if err := os.Chmod(tempFile.Name(), perm); err != nil {
-		return fmt.Errorf("error changing mode of temp file: %v", err)
-	}
-
-	if err := os.Rename(tempFile.Name(), filename); err != nil {
-		return fmt.Errorf("error moving temp file %q to %q: %v", tempFile.Name(), filename, err)
-	}
-
-	mustRemove = false
-	return nil
-}
+//func atomicWriteFile(filename string, data []byte, perm os.FileMode) error {
```

Comment:
atomicWriteFile can remove not comment?

Reply:
Looked like someone did some work to create it. Maybe someone needs it for something at some point in time.

Reply:
sure i get it

Reply:
Decided you were right. It's Git, anyone can go back to a version that had the function and copy it.